addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1177,6 +1358,129 @@ var SnippetEditor = Widget.extend({
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onSendBackClick(ev) {
+        ev.stopPropagation();
+        const rowEl = this.$target[0].parentNode;
+
+        // Finding the minimum z-index.
+        let minZindex = Number.MAX_SAFE_INTEGER;
+        for (const column of rowEl.children) {
+            minZindex = Math.min(minZindex, column.style.zIndex);
+        }
+
+        // While the minimum z-index is not 0, it is OK to decrease the column z-index.
+        // Otherwise, set it to the minimum and increase the other ones.
+        if (minZindex > 0) {
+            this.$target[0].style.zIndex = minZindex - 1;
+        } else {
+            for (const column of rowEl.children) {
+                column.style.zIndex++;
+            }
+            this.$target[0].style.zIndex = 0;
+        }
+    },
+    /**
+     * Called when the "bring to front" overlay button is clicked.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onBringFrontClick(ev) {
+        ev.stopPropagation();
+        const rowEl = this.$target[0].parentNode;
+
+        // Finding the maximum z-index.
+        let maxZindex = Number.MIN_SAFE_INTEGER;
+        for (const column of rowEl.children) {
+            maxZindex = Math.max(maxZindex, column.style.zIndex);
+        }
+
+        // While the maximum z-index is not 1040, it is OK to increase the column z-index.
+        // Otherwise, set it to the maximum and decrease the other ones.
+        if (maxZindex < 1040) { // The overlay z-index is 1040 so it is the maximum allowed.
+            this.$target[0].style.zIndex = maxZindex + 1;
+        } else {
+            for (const column of rowEl.children) {
+                column.style.zIndex--;
+            }
+            this.$target[0].style.zIndex = 1040;
+        }
+    },
+    /**
+     * Called when the mouse is moved to place a column in a grid.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onDragMove(ev) {
+        const column = this.$target[0];
+        const rowEl = column.parentNode;
+
+        // Computing the rowEl position.
+        const rowElTop = rowEl.getBoundingClientRect().top + document.documentElement.scrollTop;
+        const rowElLeft = rowEl.getBoundingClientRect().left;
+
+        // Getting the column dimensions.
+        const borderWidth = parseFloat(window.getComputedStyle(column).borderWidth);
+        const columnHeight = column.clientHeight + 2 * borderWidth;
+        const columnWidth = column.clientWidth + 2 * borderWidth;
+        const columnMiddle = columnWidth / 2;
+
+        // Placing the column where the mouse is.
+        const top = ev.pageY - rowElTop;
+        const bottom = top + columnHeight;
+        let left = ev.pageX - rowElLeft - columnMiddle;
+
+        // Horizontal overflow.
+        if (left < 0) {
+            left = 0;
+        } else {
+            left = Math.min(left, rowEl.clientWidth - columnWidth);
+        }
```

Comment:
Write that kind of stuff on one line: `left = left < 0 ? 0 : Math.min(...)`. Although here it seems like a use for a call to `confine` no?